### ISSUE TITLE
no-trailing-punctionation rule being ignored

### ISSUE THREAD

**Issue by ksnyde** [DavidAnson/markdownlint#174]
in my `.markdownlint.yaml` file (in repo root) I have several rules (see below) which all work but I wanted to modify `no-trailing-punctuation` to allow both "!" and "?"

```yaml
no-duplicate-header: false
line-length: false
no-inline-html: false
no-space-in-links: false
no-trailing-punctuation: ".,;:"
```

But as many combinations as I try I can't get this configuration adjustment to take hold. Also of note, I am using the `vs-code` environment to surface these warnings and so therefore possibly this is actually an error with the [`markdownlint` extension by David Anson](https://github.com/DavidAnson/vscode-markdownlint). I will raise this issue there too as I'm really not sure how to distinguish between the two.

**Comment by DavidAnson** [comment]
`punctuation` is a parameter to the rule, as seen here in JSON: https://github.com/DavidAnson/markdownlint/blob/2710c375b3c9b9b6531d546b667f50f7d0c956f2/test/heading_trailing_punctuation_customized.json

You need to set it as a parameter in YAML as seen here for `indent`: https://github.com/DavidAnson/markdownlint/blob/master/test/config/config-grandparent.yaml

If that’s not clear, I can give a better example when I’m not on my phone.

**Comment by ksnyde** [comment]
Many thanks @DavidAnson, that is clear. So for others who may end up here, in YAML I would configure this rule as:

```yml
no-duplicate-header: false
line-length: false
no-inline-html: false
no-space-in-links: false
no-trailing-punctuation:
  punctuation: ".,;:"
```

**Comment by DavidAnson** [comment]
Yep!
